cfme/fixtures/vporizer.py

```diff
@@ -0,0 +1,101 @@
+max_cpu_usage_rate_average__data_range = 8000
+max_mem_usage_absolute_average__data_range = 100
+
+
+def gen_vpor_values():
+    return vpor_values_pattern.format(
+        random(), random() * cpu_usagemhz_rate_average__data_range,
+        random() * 100, random() * derived_memory_used__data_range,
+        random(), random() * max_cpu_usage_rate_average__data_range,
+        random() * 100, random() * max_mem_usage_absolute_average__data_range
+    )
+
+
+@pytest.yield_fixture(scope='module')
+def vporizer(appliance):
+
+    """Grabbing vim_performance_operating_ranges table data for nodes and projects.
+    In case that no such data exists, inserting fake rows"""
+
+    db = appliance.db
+    vpor = db['vim_performance_operating_ranges']
+
+    container_nodes = db['container_nodes']
+    container_pods = db['container_groups']
+    container_projects = db['container_projects']
+    created_at = datetime.now()
+    vpor_data_list = []
+
+    for table, resource_type in zip(
+        (container_nodes, container_projects, container_pods),
+        ('ContainerNode', 'ContainerProject', 'ContainerGroup')
+    ):
+
+        for resource in db.session.query(table).all():
+            def get_resource_vpor_data():
+                return vpor.__table__.select().where(
+                    (vpor.resource_id == resource.id) &
+                    (vpor.resource_type == resource_type)
+                ).execute().fetchone()
+            # Checking whether values has already inserted to the table for this resource.
+            # Should be True only in case that the appliance age is greater than 24h
+            # if there isn't such row, insert...
+            if not get_resource_vpor_data():
+                vpor.__table__.insert().values(
+                    resource_type=resource_type, resource_id=resource.id,
+                    created_at=created_at, days=30, updated_at=datetime.now(),
+                    time_profile_id=1, values=gen_vpor_values()
+                ).execute()
+            # now, collecting the values from the table
+            resource_vpor_data = get_resource_vpor_data()
+            vpor_values_match = re.search(vpor_values_pattern.replace('{}', '([\d\.]+)'),
+                                          resource_vpor_data['values'])
+            if not vpor_values_match:
+                raise Exception('Could not parse VPOR values from row: values={}'
+                                .format(resource_vpor_data['values']))
+            vpor_values = [float(v) for v in vpor_values_match.groups()]
+            vpor_data_list.append(
+                vpor_data_instance(
+                    resource_vpor_data['resource_type'],
+                    resource_vpor_data['resource_id'], resource.name,
+                    vpor_values[0], vpor_values[1], vpor_values[2], vpor_values[3]
+                )
+            )
+
+    yield vpor_data_list
+
+    vpor.__table__.delete().where(vpor.created_at == created_at).execute()
```

Comment:
This is a bit of a nitpick: Not sure if that can be done, but it would be better to grab the IDs of the created entries and use them in this query to specifically reference those. I think the query to create the entries should somehow return the created entries with their IDs set. `created_at` does not have the uniqueness constraint so it could theoretically also reference some entries that were created by other process.